### PR TITLE
Read ctlogs via crt.sh

### PR DESCRIPTION
**Pull request by yschimke**
Read certificate logs to see all issued certificates, rather than those served by the host.

```
$ cft --host www.squareup.com --ctlogs
CN: 	www.squareup.com
...
OCSP status: GOOD

Certificate Issuers:
Entrust Certification Authority - L1K
	*.squareup.com	Valid: 	2020-11-13T22:10:42Z..2021-11-30T22:10:41Z (10 months)
	*.squareup.com	Valid: 	2020-11-13T22:10:42Z..2021-11-30T22:10:41Z (10 months)
Entrust Certification Authority - L1M
	www.squareup.com	Valid: 	2020-07-06T19:26:42Z..2021-07-05T19:26:40Z (5 months)
	www.squareup.com	Valid: 	2020-08-27T22:13:48Z..2021-08-26T22:13:47Z (7 months)
	www.squareup.com	Valid: 	2020-07-06T19:26:42Z..2021-07-05T19:26:40Z (5 months)
	www.squareup.com	Valid: 	2020-08-27T22:13:48Z..2021-08-26T22:13:47Z (7 months)
```

### CODE REVIEW

**Comment by yschimke** [comment]
Curious on your thoughts of showing ctlogs and fetching via SQL...

File: certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/ct/crt.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+fun ByteString.decodeCertificatePem(): Certificate {
+  return CertificateAdapters.certificate.fromDer(this)
+}
+
+suspend fun connectToCrtShDb(): PostgresqlConnection {
+  val conf = PostgresqlConnectionConfiguration.builder()
+    // Failing on IPv6
+    // .host("crt.sh")
+    .host("91.199.212.73")
+    .port(5432)
```
Comment (oldergod) [inline certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/ct/crt.kt:67]: Where do those come from?

**Review by oldergod — APPROVED** [review]
It's pretty cool!

If the TODO are to live an unlimited period of time, could we properly ponctuate them?

Also some copyrights are missing I think.

File: certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/certificateOutput.kt [hunk 2]
```diff
@@ -88,6 +75,23 @@ fun Certificate.prettyPrintCertificate(
...
+    periodLeft.years >= 1 -> " (${periodLeft.years} years)"
+    periodLeft.months >= 1 -> " (${periodLeft.months} months)"
+    periodLeft.days < 20 -> Ansi.AUTO.string(" (@|yellow $periodLeft days|@)")
+    else -> " (${periodLeft.days})"
+  }
+  val validityString = "Valid: \t${
+    ofEpochMilli(notBefore)
+  }..${
+    ofEpochMilli(notAfter)
+  }$periodLeftString"
```
[the remark this answers is not in the document]
Comment (yschimke) [reply certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/certificateOutput.kt:91]: That probably implies cleaning this up :)

**Comment by yschimke** [comment]
> It's pretty cool!
> 
> If the TODO are to live an unlimited period of time, could we properly ponctuate them?
> 
> Also some copyrights are missing I think.

I wanted feedback before I worried about tests, will do another pass.  It's also annoying that IPv6 connections fail and having a custom tcp/ssl connection code in the r2dbc code means you can override the Dns :(

File: certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/ct/crt.kt (hunk 1 above)
Comment (yschimke) [reply certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/ct/crt.kt:67]: A snapshot of Dns at that point in time.  Port of the postgres port.  I will probably check in with the correct host and look at how to fix cleanly.

**Comment by yschimke** [comment]
@oldergod I ended up moving code around after merge conflicts, but otherwise addressed most comments.

**Pull request merged**